week7-dom/index.html

```diff
@@ -0,0 +1,193 @@
+---
+layout: default
+permalink: /week7-dom/
+---
+<h2>Quiz 7 DOM: events and callback function</h2>
+
+<h4>Please copy and paste the code to the <a href="https://alpha.editor.p5js.org/">p5 web editor</a> to do the following questions.</h4>
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/rym0wT6R" width=200 height=200></iframe>
+  <br/>
+
+  1. Write a program that implements the above sketch. First, create a <code>p</code> element. Then use <code>.positon()</code> function to update the element's x, y positon. Then use <code>.html</code> function to update its content. In the end, use <code>.style</code> function to change its background color and add paddings to the element.
+</p>
+
+<script type="text/p5" data-preview-width="400">
+// Please copy and paste the code to the p5 web editor to do the following questions.
+var p;
+var x = 50;
+var y = 50;
+
+function setup() {
+  // create a p element.
+  p = __________('p', 'Waiting.');  
+}
+
+function draw() {
+  // set element p's x, y position
+  p._______(x, y);
+  
+  // update element p's x, y position
+  x = x + Math.round(random(-1, 1));
+  y = y + Math.round(random(-1, 1));
+  
+  // update element p's content
+  p.________('x: ' + x + '<br />y: ' + y );
+  
+  // change element p's background color
+  p.________('________', '________');
+  
+  // add paddings to element p
+  p.________('________', '________');
+}
+</script>
+
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/rJLDPApA" width=200 height=270></iframe>
+  <br/>
+
+  2. Write a program that implements the above sketch. Create a <code>button</code> element. Attach a callback function to the button's <code>mousePressed</code> event. In the callback function, give the canvas a random background color. Similarlly, create an <code>input</code> element to update some text on the canvas, and create a <code>slider</code> element to control the size of an ellipse.</p>
```

Comment:
I wonder if this is too much to jump into all at once.  Maybe just a sketch that uses a button and a callback. And then a second question to add other features?